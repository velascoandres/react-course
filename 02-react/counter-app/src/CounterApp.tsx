import React, { MouseEventHandler, useState } from 'react';

export interface ICounterAppProps {
	value: number;
}

export type MouseEHandlerFact = (...args: any) => MouseEventHandler;


const CounterApp = ({ value = 10 }: ICounterAppProps) => {


	const [counter, setCounter] = useState<number>(value);


	// handleReset
	const handleReset: MouseEventHandler = (e) => setCounter(value);
	// handleAdd Factory
	const handleAddFactory: MouseEHandlerFact = (numero: number) => {
		return (e: React.MouseEvent) => {
			// Actualizamos el estado del componente
			const updatedCounter = counter + numero;
			// setCounter( updatedCounter );
			// Otra forma de actualizar el estado
			setCounter((oldCounter) => oldCounter + numero);
		};
	};

	return (
		<div>
			<h1>Counter App</h1>
			<h2>{counter}</h2>

			<button onClick={handleAddFactory(1)}>
				+1
			</button>
			<button onClick={handleReset}>
				Reset
			</button>
			<button onClick={handleAddFactory(-1)}>
				-1
			</button>
			
		</div>
	);
}

export default CounterApp;
