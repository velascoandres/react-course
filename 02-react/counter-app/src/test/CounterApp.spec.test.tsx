import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';


describe('Pruebas con el componente <CounterApp />', () => {
    
    test('Mostrar <CounterApp /> correctamante', () => {
        
        const wrapper = shallow(<CounterApp />);

        expect(wrapper).toMatchSnapshot();

    });


    test('Debe mostrar el valor de 100', () => {
        
        const valor = 100;

        const wrapper = shallow(<CounterApp value={valor} />);

        const numero = wrapper.find('h2').text().trim();

        expect(numero).toBe(valor.toString());

    });
    


});
