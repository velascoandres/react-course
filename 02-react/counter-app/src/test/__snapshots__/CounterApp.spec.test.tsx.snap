// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <CounterApp /> Mostrar <CounterApp /> correctamante 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
